featuretools/primitives/options_utils.py

```diff
@@ -0,0 +1,234 @@
+    else:
+        def variable_filter(f):
+            return passes_ignore_filter(f)
+    return variable_filter
+
+
+def _groupby_filter_generator(options):
+    def passes_include_groupby_filter(f):
+        return (isinstance(f, IdentityFeature) and
+                (f.entity.id in options['include_groupby_variables'] and
+                f.variable.id in options['include_groupby_variables'][f.entity.id]))
+
+    def passes_ignore_groupby_filter(f):
+        return (isinstance(f, IdentityFeature) and (
+                f.entity.id not in options['ignore_groupby_variables'] or
+                (f.entity.id in options['ignore_groupby_variables'] and
+                    f.variable.id not in options['ignore_groupby_variables'][f.entity.id])))
+
+    if 'include_groupby_variables' in options and 'ignore_groupby_variables' in options:
+        def groupby_filter(f):
+            return passes_include_groupby_filter(f) or \
+                (f.entity.id not in options['include_groupby_variables'] and
+                    passes_ignore_groupby_filter(f))
+    elif 'include_groupby_variables' in options:
+        def groupby_filter(f):
+            return passes_include_groupby_filter(f) or \
+                f.entity.id not in options['include_groupby_variables']
+    elif 'ignore_groupby_variables' in options:
+        def groupby_filter(f):
+            return passes_ignore_groupby_filter(f)
+    else:
+        def groupby_filter(f):
+            return (isinstance(f, IdentityFeature))
+    return groupby_filter
+
+
+def ignore_entity_for_primitive(options, entity, groupby=False):
+    # This logic handles whether given options ignore an entity or not
+    def should_ignore_entity(option):
+        if 'include_entities' in option and \
+                entity.id not in option['include_entities'] or \
+                groupby and 'include_groupby_entities' in option and \
+                entity.id not in option['include_groupby_entities']:
+            return True
+        elif entity.id in option['ignore_entities'] or \
+                groupby and 'ignore_groupby_entities' in option and \
+                entity.id in option['ignore_groupby_entities']:
+            return True
+        else:
+            return False
+    return any([should_ignore_entity(option) for option in options])
+
+
+def filter_groupby_matches_by_options(groupby_matches, options):
+    return filter_matches_by_options([(groupby_match, ) for groupby_match in groupby_matches],
+                                     options,
+                                     groupby=True)
+
+
+def filter_matches_by_options(matches, options, groupby=False):
+    generator = _groupby_filter_generator if groupby else _variable_filter_generator
+    if len(options) > 1:
+        def is_valid_match(match):
+            variable_filter = [generator(option) for option in options]
+            if all([variable_filter[i](match[i]) for i in range(len(variable_filter))]):
```

Comment:
would it be cleaner to use zip here? 

`all([vf(m) for vf, m in zip(variable_filter, match))`